src/rules/constructorArgumentsCopy.ts

```diff
@@ -0,0 +1,67 @@
+const goodCode = `
+pragma solidity 0.4.4;
+
+
+contract C {
+    constructor(uint a_);
+}
+`;
+const badCode = `
+pragma solidity 0.4.4;
+
+
+contract C {
+    constructor(uint a);
+}
+`;
+const META = {
+    ruleId: 'constructor-arguments-unserscored',
+    type: 'miscellaneous',
+
+    docs: {
+        description: 'All contract constructor arguments should be suffixed or prefixed with an underscore.',
+        category: 'Miscellaneous',
+        examples: {
+            good: [
+                {
+                    description: 'Underscore suffix on constructor arguments',
+                    code: goodCode,
+                },
+            ],
+            bad: [
+                {
+                    description: 'Missing underscore suffix on constructor arguments',
+                    code: badCode,
+                },
+            ],
+        },
+    },
+
+    isDefault: false,
+    recommended: false,
+    defaultSetup: 'warn',
+
+    schema: null,
+};
+
+class UnderscoredConstructorArguments {
+    private ruleId: string;
+    private reporter: any;
+    private meta: Record<string, any>;
+    constructor(reporter: any) {
+        this.ruleId = META.ruleId;
+        this.reporter = reporter;
+        this.meta = META;
+    }
+    FunctionDefinition(ctx: any) {
+        if (!ctx.isConstructor) {
+            return;
+        }
+        ctx.parameters.forEach((param: any) => {
+            if (param.name[param.name.length - 1] !== '_' && param.name[0] !== '_') {
+                this.reporter.error(param, this.ruleId, 'No suffix at the end of constructor argument');
+            }
+        });
+    }
+}
+module.exports = { UnderscoredConstructorArguments, META };
```

Comment:
Not a hard requirement, but I think we can avoid using outdated commonjs syntax if we:
- Export required objects directly using es module syntax, eg `export const META`
- (Maybe) remove [`esModuleInterop`](https://www.typescriptlang.org/tsconfig#esModuleInterop) from tsconfig